apis/smartweather.js

```diff
@@ -0,0 +1,433 @@
+
+		let weather = {};
+		weather.forecasts = [];
+		let that = this;
+		weather.report = that.currentReport;
+		callback(null, weather);
+		
+		// Save the state after updating plugin state so we don't
+		// delay the update
+		this.save(that.currentReport);
+	}
+
+	// Calculate Wet Bulb Temperature
+	// TODO: May want to move this to converter file
+	getWetBulbTemperature(dryBulbTemperature, relativeHumidity)
+	{
+		let T = dryBulbTemperature;
+		let rh = relativeHumidity;
+
+		let c1 = 0.152;
+		let c2 = 8.3136;
+		let c3 = 0.5;
+		let c4 = 1.6763;
+		let c5 = 0.00391838;
+		let c6 = 1.5;
+		let c7 = 0.0231;
+		let c8 = 4.686;
+
+		let Tw = T * Math.atan(c1 * Math.pow((rh + c2), c3)) +
+			Math.atan(T+rh) - Math.atan(rh-c4) + 
+			c5 * Math.pow(rh, c6) * Math.atan(c7 * rh) - c8;
+
+		return Tw;
+	}
+
+
+	// Map Smart Weather precipitation values to Eve Condition Categories
+	getConditionCategory(precipitationType, detail = false)
+	{
+	// Tempest: 0 = none, 1 = rain, 2 = hail, 3 = rain + hail (experimental)
+	
+	// Eve (simple): 0 = clear; 3 = snow; 2 = rain; 1 = overcast
+	// Eve (detailed): 0 = clear; 1 = Few clouds;2 = Broken clouds;3 =  Overcast;
+	//   4 = Fog; 5 = Drizzle; 6 = Rain; 7 = Hail; 8 = Snow; 9 = Severe weather
+
+		if (precipitationType == 1) {
+			// Rain
+			return detail ? 6 : 2;
+		} else if (precipitationType == 2) {
+			// Hail
+			return detail ? 7 : 2;
+		} else if (precipitationType == 3){
+			// Rain + Hail (return as hail)
+			return detail ? 7 : 2;
+		} else {
+			// 0 = Clear
+			return 0;
+		}
+	}
+
+	// Relying on the update happening every minute
+	getHourlyAccumulatedRain(mmOfRainInLastMinute)
+	{
+		let that = this;
+		
+		that.rainAccumulation[that.rainAccumulationMinute] = mmOfRainInLastMinute;
+	
+		// Can't use util/converter function getRainAccumulated(array[values][field], field)
+		// as it takes two dimension array, and SmartWeather only needs one dimension, also
+		// it may convert the sum to int and this needs to be float.
+		var accumulation = 0.0;
+		for (var i = 0; i < 60; i++) {
+			accumulation += parseFloat(that.rainAccumulation[i]);
+		}
+	
+		that.rainAccumulationMinute++;
+		if (that.rainAccumulationMinute > 59) {
+			that.rainAccumulationMinute = 0;
+		}
+		this.log.debug("getHourlyAccumulatedRain last minute: " + mmOfRainInLastMinute + " last hour: " + accumulation);
+		return accumulation;
+	}
+
+	// Assume that zero battery level is 2.6v
+	getBatteryPercent(batteryVoltage)
+	{
+		return (batteryVoltage * 100 - 260);
+	}
+	
+	// Tempest battery ranges from 2.355 (low) to 2.8 (full)
+	// https://help.weatherflow.com/hc/en-us/articles/360048877194-Solar-Power-Rechargeable-Battery
+	// Assume "low" to be 5%, so "zero" battery level is approx. 2.11v
+	getTempestBatteryPercent(batteryVoltage)
+	{
+		var percent = (batteryVoltage * 100.0 - 211.0)/0.69;
+		return (percent > 100) ? 100 : (percent < 0.0) ? 0 : percent;
+	}
+	
+
+	// API: https://weatherflow.github.io/SmartWeather/api/udp/v119/
```

Comment:
https://weatherflow.github.io/SmartWeather/api/udp/v119/ is dead, it redirects elsewhere:
```suggestion
	// API: https://weatherflow.github.io/Tempest/api/udp/v171/
```